Simplify utils.get_value

taion opened the pull request:
This is good for like a 10% speedup in the benchmark. It's not going to make a big difference, but it's a free win.

taion committed:
Simplify utils.get_value

taion committed:
fix

taion committed:
AttributeError can't get thrown here

sloria commented on marshmallow/utils.py:
```diff
@@ -338,7 +338,7 @@ def _get_value_for_keys(obj, keys, default):
 def _get_value_for_key(obj, key, default):
     try:
         return obj[key]
-    except (TypeError, AttributeError):
```
Based on the failing Travis build, it looks like this might be necessary to support custom iterables that don't explicitly implement `__getitem__` in Python 2.

sloria reviewed (changes requested):
Once the failing Travis build is addressed, this should be good to merge.

taion committed:
actually, it can be

taion commented:
updated and fixed – i didn't realize that python2 throws `AttributeError` when subscripting into un-subscriptable new-style class instances rather than `TypeError`
